packages/nextjs/test/utils/withSentry.test.ts

```diff
@@ -0,0 +1,92 @@
+import * as Sentry from '@sentry/node';
+import * as utils from '@sentry/utils';
+import { NextApiHandler, NextApiRequest, NextApiResponse } from 'next';
+
+import { AugmentedNextApiResponse, withSentry, WrappedNextApiHandler } from '../../src/utils/withSentry';
+
+const FLUSH_DURATION = 200;
+
+async function sleep(ms: number): Promise<void> {
+  await new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function callWrappedHandler(wrappedHandler: WrappedNextApiHandler, req: NextApiRequest, res: NextApiResponse) {
+  await wrappedHandler(req, res);
+
+  // Within the wrapped handler, we await `flush()` inside `res.end()`, but nothing awaits `res.end()`, because in its
+  // original version, it's sync (unlike the function we wrap around it). The original does actually *act* like an async
+  // function being awaited - subsequent steps in the request/response lifecycle don't happen until it emits a
+  // `prefinished` event (which is why it's safe for us to make the switch). But here in tests, there's nothing past
+  // `res.end()`, so we have to manually wait for it to be done.
+  await sleep(FLUSH_DURATION);
+  while (!res.finished) {
+    await sleep(100);
+  }
+}
+
+// We mock `captureException` as a no-op because under normal circumstances it is an un-awaited effectively-async
+// function which might or might not finish before any given test ends, potentially leading jest to error out.
+const captureExceptionSpy = jest.spyOn(Sentry, 'captureException').mockImplementation(jest.fn());
```

Comment:
Sorry - I'm not clear what you mean here, exactly. `jest.fn()` _is_ a no-op, and spying exists precisely in order to be able to test against whether and how many times said function has been called. So the existing code is doing exactly what you're hoping for, is it not? Or am I misunderstanding you?

Reply:
Do we want it to be a no-op? We definitely don't want it to send events to Sentry, but we may want to set another behavior for the tests. This is related to the order of actions in the tests below, we can ignore the comment if the order isn't something we want to consider.

Reply:
Yes, it's fine for it to be a no-op, because it's not what we're testing here. What we're testing is a function which calls it, and so that's what we need to be sure of - that it gets called, and that's exactly what spies are designed to do.